test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -947,18 +944,22 @@ def test_terminal_control_hook_name(self):
         """
         # Create hook
         hook_name = "h%s%sd" % (self.bold, self.red)
-        a = {'event': "execjob_begin", 'enabled': 'True', 'debug': 'True'}
-        try:
-            rv = self.server.create_hook(hook_name, a)
-        except PbsManagerError:
+        cmd = ['qmgr', '-c', 'create hook %s' % hook_name]
+        ret = self.du.run_cmd(self.server.hostname, cmd, sudo=True)
+        if ret['rc'] != 0:
             # Delete the hook first then create the hook
-            self.server.manager(MGR_CMD_DELETE, HOOK, id=hook_name)
-            rv = self.server.create_hook(hook_name, a)
-        self.assertTrue(rv)
+            cmd = ['qmgr', '-c', 'delete hook %s' % hook_name]
+            ret = self.du.run_cmd(self.server.hostname, cmd, sudo=True)
+            self.assertEqual(ret['rc'], 0)
+            cmd = ['qmgr', '-c', 'create hook %s' % hook_name]
+            ret = self.du.run_cmd(self.server.hostname, cmd, sudo=True)
+            self.assertEqual(ret['rc'], 0)
         hook_name_esc = "h%s%sd" % (self.bold_esc, self.red_esc)
         self.check_print_list_hook(hook_name, hook_name_esc)
         # Delete the hook
-        self.server.manager(MGR_CMD_DELETE, HOOK, id=hook_name, expect=True)
+        cmd = ['qmgr', '-c', 'delete hook %s' % hook_name]
```

Comment:
why did you change it to use run_cmd?

Reply:
server.manager does not seem to work well with ANSI escape codes used for terminal control characters.